roles/kubernetes/preinstall/templates/resolved.conf.j2

```diff
@@ -1,21 +1,15 @@
 [Resolve]
-{% if dns_early is sameas true and dns_late is sameas false %}
-#DNS=
-{% else %}
+{% if not dns_early and dns_late %}
```

Comment:
/lgtm

Notes: 

There are some differences between using `if dns_late is sameas false ` and only `if dns_late`.
If the dns_late is not not defined. The behavior is different. 
But it's defined at the https://github.com/kubernetes-sigs/kubespray/blob/v2.23.1/roles/kubernetes/preinstall/defaults/main.yml#L7, so it's OK.

Reply:
I'm aware, but we shouldn't handle that in jinja manifests (`is defined` is bad), but in the defaults/vars. That's the same problem as using `default` everywhere. We have two dedicated facilities for avoiding this (vars and defaults), let's use them !